frontend/src/__tests__/UserLogin.test.jsx

```
@@ -0,0 +1,250 @@
+    // Wait for the error message to appear
+    await waitFor(() => {
+      const errorElement = screen.getByText('Something went wrong. Please try again.');
+      expect(errorElement).toBeInTheDocument();
+      expect(console.error).toHaveBeenCalled();
+    });
+  });
+
+  test('redirects to Google OAuth when clicking Google login button', () => {
+    // Mock window.location
+    const locationAssignMock = vi.fn();
+    delete window.location;
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)